parking_permits_app/models.py

```diff
@@ -0,0 +1,386 @@
+    )
+    holder = models.ForeignKey(
+        Customer,
+        verbose_name=_("Holder"),
+        on_delete=models.PROTECT,
+        related_name="vehicles_holder",
+    )
+
+    class Meta:
+        db_table = "vehicle"
+        verbose_name = _("Vehicle")
+        verbose_name_plural = _("Vehicles")
+
+    def __str__(self):
+        return "%s - %s, %s, %s, %s" % (
+            self.id,
+            self.type,
+            self.category,
+            self.manufacturer,
+            self.model,
+        )
+
+
+class DrivingClass(models.Model):
+    id = models.UUIDField(
+        primary_key=True, unique=True, editable=False, default=uuid.uuid4
+    )
+    class_name = models.CharField(
+        _("Driving class name"), max_length=32, blank=False, null=False
+    )
+    identifier = models.CharField(
+        _("Driving class identifier"), max_length=32, blank=False, null=False
+    )
+
+    class Meta:
+        db_table = "driving_class"
+        verbose_name = _("Driving class")
+        verbose_name_plural = _("Driving classes")
+
+    def __str__(self):
+        return "%s - %s - %s" % (self.id, self.class_name, self.identifier)
+
+
+class DrivingLicence(models.Model):
+    id = models.UUIDField(
+        primary_key=True, unique=True, editable=False, default=uuid.uuid4
+    )
+    customer = models.ForeignKey(
+        Customer, verbose_name=_("Customer"), on_delete=models.PROTECT
+    )
+    driving_classes = models.ManyToManyField(DrivingClass)
+    valid_start = models.DateTimeField(_("Valid start"))
+    valid_end = models.DateTimeField(_("Valid end"))
+    active = models.BooleanField(null=False, default=True)
+
+    class Meta:
+        db_table = "driving_licence"
+        verbose_name = _("Driving licence")
+        verbose_name_plural = _("Driving licences")
+
+    def __str__(self):
+        return "%s - %s, active: %s" % (self.id, self.customer, self.active)
+
+
+class ContractType(models.Model):
+    CONTRACT_TYPES = (
+        ("FIXED_PERIOD", _("Fixed period")),
+        ("OPEN_ENDED", _("Open ended")),
+    )
+    id = models.UUIDField(
+        primary_key=True, unique=True, editable=False, default=uuid.uuid4
+    )
+    contract_type = models.CharField(
+        _("Contract type"),
+        max_length=16,
+        blank=False,
+        null=False,
+        choices=CONTRACT_TYPES,
+    )
+    month_count = models.IntegerField(_("Month count"), blank=True, null=True)
+
+    class Meta:
+        db_table = "contract_type"
+        verbose_name = _("Contract type")
+        verbose_name_plural = _("Contract types")
+
+    def __str__(self):
+        return "%s - %s, months: %s" % (self.id, self.contract_type, self.month_count)
+
+
+class Product(models.Model):
+    CONTRACT_TYPES = (
+        ("FIXED_PERIOD", _("Fixed period")),
+        ("OPEN_ENDED", _("Open ended")),
+    )
+    id = models.UUIDField(
+        primary_key=True, unique=True, editable=False, default=uuid.uuid4
+    )
+    common_id = models.UUIDField(
```

Comment:
If this is for Talpa anyway, wouldn't it be better to name this field `talpa_product_id` or even `shopping_cart_provider_product_id` or something similar instead of `common_id`? `common_id` doesn't mean anything to me in this context and if there was no comment (`# Talpa ID`), I'd have no idea what this field is about.

Reply:
Ok, let's use `shared_product_id` here.